integrationTests/chainSimulator/bridge/executeAndBurn_test.go

```diff
@@ -259,40 +244,91 @@ func getBridgedValue(bridgeInTokens []chainSim.ArgsDepositToken, token string) (
 
-	esdtValue, err := hex.DecodeString(accountKeys[getTokenKey(token)])
-	require.Nil(t, err)
+	// get user account token data
+	esdtValue := getAccountTokenData(t, nodeHandler, account, token)
 
 	if expectedAmount.Cmp(big.NewInt(0)) == 0 {
-		require.Empty(t, esdtValue) // expect that key doesn't exist in account
+		require.Empty(t, esdtValue)            // expect that key doesn't exist in account
+		if metaDataOnUserAccount(token.Type) { // for other token types the key can exist because other wallets in shard can have the token
+			requireNoTokenDataInSysAccount(t, nodeHandler, token) // no keys in system account
+		}
 		return
 	}
 
 	esdtData := &esdt.ESDigitalToken{}
-	err = nodeHandler.GetCoreComponents().InternalMarshalizer().Unmarshal(esdtData, esdtValue)
+	err := nodeHandler.GetCoreComponents().InternalMarshalizer().Unmarshal(esdtData, esdtValue)
 	require.Nil(t, err)
+	require.NotNil(t, esdtData)
 	require.Equal(t, expectedAmount, esdtData.Value)
 	require.Equal(t, uint32(token.Type), esdtData.Type)
-	if token.Nonce > 0 {
+
+	if token.Type == core.Fungible {
+		require.Nil(t, esdtData.TokenMetaData)
+		requireNoTokenDataInSysAccount(t, nodeHandler, token) // no keys in system account
+	} else if token.Type == core.NonFungibleV2 || token.Type == core.DynamicNFT {
+		require.NotNil(t, esdtData.TokenMetaData)
+		require.Equal(t, token.Nonce, esdtData.TokenMetaData.Nonce)
+		requireNoTokenDataInSysAccount(t, nodeHandler, token) // no keys in system account
+	} else {
+		require.Nil(t, esdtData.TokenMetaData)
+
+		// get system account token data
+		esdtValue = getAccountTokenData(t, nodeHandler, esdtSystemAccount, token)
+
+		esdtData = &esdt.ESDigitalToken{}
+		err = nodeHandler.GetCoreComponents().InternalMarshalizer().Unmarshal(esdtData, esdtValue)
+		require.Nil(t, err)
+		require.NotNil(t, esdtData)
+		require.GreaterOrEqual(t, esdtData.Value.Uint64(), expectedAmount.Uint64()) // greater if other wallets have the token
+		require.Equal(t, uint32(token.Type), esdtData.Type)
+		require.NotNil(t, esdtData.TokenMetaData)
 		require.Equal(t, token.Nonce, esdtData.TokenMetaData.Nonce)
 	}
 }
 
+func getAccountTokenData(
+	t *testing.T,
+	nodeHandler process.NodeHandler,
+	account string,
+	token chainSim.ArgsDepositToken,
+) []byte {
+	accountKeys, _, err := nodeHandler.GetFacadeHandler().GetKeyValuePairs(account, dataApi.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, accountKeys)
+
+	esdtValue, err := hex.DecodeString(accountKeys[getTokenKey(token)])
+	require.Nil(t, err)
+
+	return esdtValue
+}
+
+func requireNoTokenDataInSysAccount(
+	t *testing.T,
+	nodeHandler process.NodeHandler,
+	token chainSim.ArgsDepositToken,
+) {
+	accountKeys, _, err := nodeHandler.GetFacadeHandler().GetKeyValuePairs(esdtSystemAccount, dataApi.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, accountKeys)
+	require.Empty(t, accountKeys[getTokenKey(token)])
+}
+
+func metaDataOnUserAccount(esdtType core.ESDTType) bool {
```

Comment:
This func just checks esdt type, could be renamed

Reply:
It is used in the context of checking metadata. Checks if metadata is on user account based on esdt type. 
Any suggestions of name?